### ISSUE TITLE
voluptuous.error.MultipleInvalid: extra keys not allowed @ data['schema']

### ISSUE THREAD

**Issue by petterl** [icanos/hassio-plejd#61]
The generated discover event seem to not be interpreted correctly by mqtt integration becouse of an extra schema data. This on a newly installed hassio with mosquitto as addin but mqtt as normal config not integration with setting:

```
mqtt:
  broker: 'core-mosquitto.local.hass.io'
  port: 1883
  username: 'plejd'
  password: '****'
  discovery: true
  discovery_prefix: homeassistant
  birth_message: 
    topic: 'hass/status'
    payload: 'online'
  will_message: 
    topic: 'hass/status'
    payload: 'offline'
```
The error in the logs are:
```
2020-02-09 12:13:45 ERROR (MainThread) [homeassistant.util.logging] Exception in async_discover when dispatching 'mqtt_discovery_new_switch_mqtt': ({'schema': 'json', 'name': 'Strömbrytare staket', 'unique_id': 'light.plejd.strömbrytarestaket', 'state_topic': 'homeassistant/switch/plejd/12/state', 'command_topic': 'homeassistant/switch/plejd/12/set', 'optimistic': False, 'brightness': 'false', 'device': {'identifiers': 'D26BDF1DF6C1_12', 'manufacturer': 'Plejd', 'model': 'WPH-01', 'name': 'Strömbrytare staket', 'sw_version': '0.3.1'}, 'platform': 'mqtt'},)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/mqtt/switch.py", line 81, in async_discover
    config = PLATFORM_SCHEMA(discovery_payload)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/usr/local/lib/python3.7/site-packages/voluptuous/schema_builder.py", line 432, in validate_mapping
    raise er.MultipleInvalid(errors)
voluptuous.error.MultipleInvalid: extra keys not allowed @ data['schema']
```

**Comment by jzee** [comment]
that's a plejd switch (WPH-01) - I'm guessing that's expected since they are not yet supported and hence the add-on doesn't know what to do with it. I get the same log message but didn't notice any problems with it - the switch still works fine in direct communication with the dimmers, the fact that the add-on doesn't yet know how to deal with it doesn't appear to cause any problems

**Comment by icanos** [comment]
Yes, I haven't yet gotten to include the WPH-01. I'll try to look at it as soon as possible :) but until then, you should be able to ignore that error.

**Comment by StigGHelmer** [comment]
I get the same error message except as above for each of the channels on my REL-2 (which has been earlier reported to work) on a RPI 3b+ both before and after the latest updates both on system side and the latest. 

Any suggestions how to proceed?

**Comment by icanos** [comment]
Yes, I have seen this and its a confirmed bug. I'm working on it and it will be released (and fixed) once I've ironed out WPH-01 and scene triggering.

**Comment by icanos** [comment]
Resolved in #64
